src/mscorlib/src/System/Environment.cs

```diff
@@ -701,20 +690,19 @@ private static IDictionary GetEnvironmentVariablesCore(EnvironmentVariableTarget
             }
             else
             {
-                throw new ArgumentException(SR.Format(SR.Arg_EnumIllegalVal, (int)target));
+                throw new ArgumentOutOfRangeException(nameof(target), target, SR.Format(SR.Arg_EnumIllegalVal, target));
```

Comment:
I will be non-trivial for somebody to reverse engineer this once they get to it. I think it is useful to keep the code the same where possible, even if it was not moved to the shared partition yet...